examples/with-thirdparty/app/src/main/java/com/supertokens/supertokensexample/LoginActivity.kt

```diff
@@ -0,0 +1,195 @@
+
+        if (data == null) {
+            Toast.makeText(baseContext, "Github sign in failed", Toast.LENGTH_SHORT).show()
+            return
+        }
+
+        val response = AuthorizationResponse.fromIntent(data)
+        val exception = AuthorizationException.fromIntent(data)
+
+        if (exception != null) {
+            exception.message?.let { msg -> Log.e("Login Activity", msg) }
+            Toast.makeText(baseContext, "Github sign in failed", Toast.LENGTH_SHORT).show()
+            return
+        }
+
+        if (response == null) {
+            Toast.makeText(baseContext, "Github sign in failed", Toast.LENGTH_SHORT).show()
+            return
+        }
+
+        val code = response.authorizationCode
+        val idToken = response.idToken
+        val state = response.state
+
+        val body = JSONObject()
+        body.put("thirdPartyId", "github")
+
+        if (idToken != null) {
+            val oauthTokens = JSONObject()
+            oauthTokens.put("id_token", idToken)
+            body.put("oAuthTokens", oauthTokens)
+        }
+
+        if (code != null && state != null) {
+            val redirectURIInfo = JSONObject()
+            redirectURIInfo.put("redirectURIOnProviderDashboard", "com.supertokens.supertokensexample://oauthredirect")
+
+            val redirectURIQueryParams = JSONObject()
+            redirectURIQueryParams.put("code", code)
+            redirectURIQueryParams.put("state", state)
+
+            redirectURIInfo.put("redirectURIQueryParams", redirectURIQueryParams)
+            body.put("redirectURIInfo", redirectURIInfo)
+        }
+
+        val requestBody = RequestBody.create(MediaType.parse("application/json; charset=utf-8"), body.toString())
+        APIManager.getInstance().retrofitService.signInUp(requestBody).enqueue(object: Callback<ResponseBody> {
+            override fun onResponse(
+                call: Call<ResponseBody>,
+                response: Response<ResponseBody>
+            ) {
+                if (response.isSuccessful) {
+                    val intent = Intent(baseContext, MainActivity::class.java)
+                    intent.flags = FLAG_ACTIVITY_NEW_TASK or FLAG_ACTIVITY_CLEAR_TASK
+                    startActivity(intent)
+                }
+            }
+
+            override fun onFailure(call: Call<ResponseBody>, t: Throwable) {
+                t.message?.let { msg -> Log.e("Login Activity", msg) }
+                Toast.makeText(baseContext, "Github sign in failed", Toast.LENGTH_SHORT).show()
+            }
+        })
+    }
+
+    private fun onGoogleResultReceived(it: ActivityResult) {
+        val task = GoogleSignIn.getSignedInAccountFromIntent(it.data)
+
+        try {
+            val account = task.result
+            val idToken = account.idToken
+
+            val body = JSONObject()
+            body.put("thirdPartyId", "google")
+
+            val oauthTokens = JSONObject()
+            oauthTokens.put("id_token", idToken)
```

Comment:
in the docs, we also pass in the access_token. Why is that missing here?

Reply:
Google does not provide an access token in native app flows, the id token is also something we specifically configure with google so that we can read it in the frontend. The user info is generally available in the result of the google login so sending the access token to the frontend would be redundant

Reply:
See if you can use the code flow instead of with id_token